### DISCUSSION TITLE
PHP Version per Directory

### DISCUSSION THREAD

**Discussion by DewlanceHosting** [leokhoa/laragon#667] [Ideas]
Please add the option to serve PHP per directory. I want to **use PHP 7.4 for /blog**, PHP 8.0 for /forum, and **PHP 7.2 for another directory.**

**Comment by kstingel** [comment]
>  I want to **use PHP 7.4 for /blog**, PHP 8.0 for /forum, and **PHP 7.2 for another directory**.  

Why would you be looking to run retired versions of PHP?

**PHP 7.2.34** (final release) was 2020-10-01 and Security Support ended on 2020-11-30  
**PHP 7.4.33** (final release) was 2022-11-03 and Security Support ended on 2022-11-30  
**PHP 8.0.30** (final release) was 2023-08-03 and Security Support ended on 2023-11-30  
Security Support for **PHP 8.1** is scheduled to end on 2024-11-24, the current release is _PHP 8.1.27_  
**PHP 8.2** will be supported until 2025-12-08, and **PHP 8.3** has support until 2026-11-26,  
although PHP 8.3 currently has issues playing nice with Apache 2.4.58 on Windows

**Comment by DewlanceHosting** [comment]
Some scripts may not work with the latest PHP version. Laragon is installed on my local computer, so I do not have issue with PHP security. I mean I am the only one who access Laragon/web server. 

Also, with the per-directory option, we won't have to worry about changing the PHP version whenever we need to work with a different PHP version.

**Reply by kstingel** [reply]
Apologies, I was thinking of public facing server situations. On a local dedicated machine, the security implications are moot.

There are cases where we need to revive a retired PHP... when porting/debugging older libraries for use in newer PHP releases.

**Comment by Pen-y-Fan** [comment]
This discussion may help:  #416
